conbench/entities/history.py

```diff
@@ -570,3 +582,67 @@ def _calculate_z_score(
         z_score = z_score * -1
 
     return z_score
+
+
+def zscore_automated_detect(df: pd.DataFrame, z_score_threshold=5.0) -> pd.DataFrame:
+    """Detect outliers and distribution shifts in historical data
+
+    Uses a z-score-based detection algorithm, taking a dataframe of the same
+    sort as `_add_rolling_stats_columns_to_df`, and returning that dataframe with
+    two additional columns:
+
+    - `is_step` (bool): Is this point the start of a new segment?
+    - `is_outlier` (bool): Is this point an outlier that should be ignored?
```

Comment:
Can you comment on how you think about using the output? 

I can imagine the obvious part where `is_outlier` would be used in `_add_rolling_stats_columns_to_df()`  to exclude those points for all calculations. I think this is what you do above in all those places where I see `df.loc[~df.is_outlier]`.

It appears as if adding `is_step` to the otuput could be controversial; this could be kind of more than we initially wanted, right?